source/common/router/upstream_request.cc

```diff
@@ -281,10 +284,23 @@ void UpstreamRequest::decodeHeaders(Http::ResponseHeaderMapPtr&& headers, bool e
     paused_for_connect_ = false;
   }
 
+  maybeHandleDeferredReadDisable();
   ASSERT(headers.get());
   parent_.onUpstreamHeaders(response_code, std::move(headers), *this, end_stream);
 }
 
+void UpstreamRequest::maybeHandleDeferredReadDisable() {
```

Comment:
this looks correct but I'd also like to see unit tests where we receive 1xx, disable, then receive headers and don't disable.

Reply:
done